### ISSUE TITLE
Read user's appearance preferences to set theme color scheme

### ISSUE THREAD

**Issue by snhasani** [bluesky-social/social-app#682]
**Is your feature request related to a problem? Please describe.**

It is about App dark mode. Right now the App doesn't read the user's appearance preferences. Because of that the user needs to switch between dark and light mode manually each day.

**Describe the solution you'd like**

We can use the `Appearance` module of `react-native` to get system appearance preferences and set it as App theme.

In both `App.web.tsx` and `App.native.tsx` we can read the current colorScheme and subscribe its changes. Something like this:

```typescript
const colorScheme = Appearance.getColorScheme()

useEffect(() => {
    const subscription = Appearance.addChangeListener(({colorScheme: mode}) => {
      rootStore.shell.setDarkMode(mode)
    })

    return () => {
      subscription.remove()
    }
  }, [rootStore?.shell])
``` 

**Describe alternatives you've considered**

The alternative solution which I recommend and like can be having a 3 mode for appearance: Dark, Light and Auto. We need a radio group element for this. Like the one we have in the content moderation modal.

<img width="369" alt="image" src="https://github.com/bluesky-social/social-app/assets/23160/59413bec-16bf-4dda-ae2d-2df0e149e8b1">

**Comment by ericvolp12** [comment]
FYI I put together a working implementation of this in https://github.com/bluesky-social/social-app/pull/688, it's not using a radio button but it cycles through system, light, and dark modes.

**Comment by ansh** [comment]
This was merged in #688
